### PR TITLE
Initial implementation of submit_multi PDU

### PR DESCRIPTION
**Pull request by VDVsx**
Tested with SMPPSim simulator, tried to not change the API much to avoid more repetition, so SM can optionally hold submit multi parameters, but any improvement suggestions welcome.
Once approach is validate I'll add more code to validate size of addresses and distribution lists as per SMPP spec.

### CODE REVIEW

**Comment by VDVsx** [comment]
@fiorix could you please take a look ? thx

File: smpp/transmitter.go [hunk 1]
```diff
@@ -135,6 +141,25 @@ func (t *Transmitter) Close() error {
...
+	AddrTON uint8
+	AddrNPI uint8
+	Address string
+	Error   pdu.Status
+}
+
+// newUnsucessDest returns a new UnsucessDest constructed from a UnSme struct
+func newUnsucessDest(p pdufield.UnSme) UnsucessDest {
+	unDest := UnsucessDest{}
+	unDest.AddrTON, _ = p.Ton.Raw().(uint8) // if there an error default value will be set
```
Comment (fiorix) [inline smpp/transmitter.go]: "if there is"...?

File: smpp/transmitter.go [hunk 2]
```diff
@@ -152,6 +177,8 @@ const (
 type ShortMessage struct {
 	Src      string
 	Dst      string
+	DstList  []string // List of destination addreses for submit multi
```
Comment (fiorix) [inline smpp/transmitter.go]: Do we want this here? Maybe we want a separate struct `ShortMessageList` or something, to keep things clean.

**Comment by fiorix** [comment]
Looks ok for the most part. I'm just unsure about the `ShortMessage` struct having mutually exclusive fields `Dst` vs these new lists. Also, have you checked for code coverage? I think the UT you added does not cover failure scenarios.

**Comment by VDVsx** [comment]
No didn't check code coverage, will do.
I'm also a bit in the fence regarding this new fields, this way we could use same API, if we create new struct just for the multi sending, some fields would need to be repeated. Another option is to wrap these new fields inside a new struct that would live inside the ShortMessage, was the later what you mean above ?

**Comment by fiorix** [comment]
I was thinking to have all separate. But I get that it will lead to a lot of code repetition.

**Comment by VDVsx** [comment]
Checked and improved the coverage now regarding the added code (before pr -> after pr):
smpp: 82.7% ->  83.1%
pdufield: 72,5% -> 80.7%

**Comment by VDVsx** [comment]
@fiorix so how should we proceed with this one ?

**Comment by fiorix** [comment]
I was about to click Merge but then I have one more question: will this also work for long messages? e.g. long message with dest list.

**Comment by VDVsx** [comment]
No, SubmitLongMsg() has own implementation I didn't change that one, so those new fields will be ignored in that case.
SubmitLongMsg and Submit() also share a lot of common code, first step would be to merge those/create new function with shared code.

**Comment by VDVsx** [comment]
ping ?

**Comment by fiorix** [comment]
Thanks! 👍

**Pull request merged**